Fails to build in GNOME Continuous

Commit 8623cb5741e149e0f4087ea01b520902190e8b90 added a C99-ism:

```
for (unsigned char i = 0; i < cap_data_len; i++) {
```

which requires -std=c99 in order to build on GCC. This was caught by the GNOME Continuous builder:

```
../lsusb.c: In function 'dump_platform_device_capability_desc':
../lsusb.c:3818:2: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
  for (unsigned char i = 0; i < cap_data_len; i++) {
  ^
../lsusb.c:3818:2: note: use option -std=c99, -std=gnu99, -std=c11 or -std=gnu11 to compile your code
Makefile:597: recipe for target 'lsusb-lsusb.o' failed
make[2]: *** [lsusb-lsusb.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

Full log: http://build.gnome.org/continuous/buildmaster/builds/2016/04/29/40/build/log-usbutils.txt

--- a/lsusb.c
+++ b/lsusb.c
@@ -3803,6 +3803,7 @@ static void dump_platform_device_capability_desc(unsigned char *buf)
 {
 	unsigned char desc_len = buf[0];
 	unsigned char cap_data_len = desc_len - 20;
+	unsigned char i;
 	if (desc_len < 20) {
 		fprintf(stderr, "  Bad Platform Device Capability descriptor.\n");
 		return;
@@ -3815,7 +3816,7 @@ static void dump_platform_device_capability_desc(unsigned char *buf)
 			buf[0], buf[1], buf[2], buf[3]);
 	printf("    PlatformCapabilityUUID    %s\n",
 			get_guid(&buf[4]));
-	for (unsigned char i = 0; i < cap_data_len; i++) {
+	for (i = 0; i < cap_data_len; i++) {
 		printf("    CapabilityData[%u]    0x%02x\n", i, buf[20 + i]);
 	}
 }

